src/rgw/rgw_sync_module_aws.cc

```diff
@@ -812,7 +812,7 @@ class RGWAWSHandleRemoteObjCBCR: public RGWStatRemoteObjCBCR {
                                                   sync_env->http_manager,
                                                   target_bucket_name, nullptr, bl, nullptr));
         }
-        if (retcode < 0) {
+        if (retcode < 0 && retcode != -ENOTEMPTY) {
```

Comment:
are you sure it's not `-EEXIST`?

Reply:
At now, the retcode is `-39(ENOTEMPTY)`, the related logs are as below:
```
2017-11-29 20:31:40.397448 7f56d2c6f700 20 cr:s=0x7f570209d560:op=0x7f5702789800:23RGWPutRawRESTResourceCRIiE: operate()
2017-11-29 20:31:40.397514 7f56d2c6f700  5 failed to wait for op, ret=-39: PUT rgwx43f7d3c24b444dd385b9eb984104193c93c-1253596042.cos.ap-chengdu.myqcloud.com/
2017-11-29 20:31:40.397536 7f56d2c6f700 20 cr:s=0x7f570209d560:op=0x7f5702789800:23RGWPutRawRESTResourceCRIiE: operate() returned r=-39
2017-11-29 20:31:40.397550 7f56d2c6f700 20 cr:s=0x7f570209d560:op=0x7f5701dd0000:25RGWAWSHandleRemoteObjCBCR: operate()
2017-11-29 20:31:40.397553 7f56d2c6f700 20 cr:s=0x7f570209d560:op=0x7f5701dd0000:25RGWAWSHandleRemoteObjCBCR: operate() returned r=-39
2017-11-29 20:31:40.397578 7f56d2c6f700 20 cr:s=0x7f570209d560:op=0x7f5701bf2000:23RGWAWSHandleRemoteObjCR: operate()
2017-11-29 20:31:40.397580 7f56d2c6f700  0 RGWStatRemoteObjCR() callback returned -39
2017-11-29 20:31:40.397581 7f56d2c6f700 20 cr:s=0x7f570209d560:op=0x7f5701bf2000:23RGWAWSHandleRemoteObjCR: operate() returned r=-39
2017-11-29 20:31:40.397586 7f56d2c6f700 20 cr:s=0x7f570209d560:op=0x7f5701ce7000:26RGWBucketSyncSingleEntryCRISs11rgw_obj_keyE: operate()
2017-11-29 20:31:40.397618 7f56d2c6f700 10 RGW-SYNC:data:sync:shard[104]:entry[samonlv:170a17b2-38c4-42e7-9ecc-fafcd3a81e39.4115.1]:bucket[samonlv:170a17b2-38c4-42e7-9ecc-fafcd3a81e3
9.4115.1]:inc_sync[samonlv:170a17b2-38c4-42e7-9ecc-fafcd3a81e39.4115.1]:entry[tt2]: failed, retcode=-39 ((39) Directory not empty) 
``` 
But to be honest, I think the reasonable retcode should be EEXIST or ERR_BUCKET_EXISTS

Reply:
oh, I see. Yeah, the `409` is ambiguous. Maybe we can somehow pass the http status code and use that here instead?